File: frontend/petcare/lib/models/adoptionRequest.dart
class AdoptionRequest {
  final int id;
  final int petId;
  final int userId;
  final String status;
  final String? message;

  AdoptionRequest({
    required this.id,
    required this.petId,
    required this.userId,
    required this.status,
    this.message,
  });

  factory AdoptionRequest.fromJson(Map<String, dynamic> json) {
    return AdoptionRequest(
      id: json['id'] as int,
      petId: json['petId'] as int,
      userId: json['userId'] as int,
      status: json['status'] as String,
      message: json['message'] as String?,
    );
  }
  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'petId': petId,
      'userId': userId,
      'status': status,
      'message': message,
    };
  }
}

File: backend/PetCare/Entities/AdoptionRequest.cs
using PetCare.Entities;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;

namespace PetCare.Entities;

public class AdoptionRequest
{
    [Key]
    public int Id { get; set; }

    [ForeignKey(nameof(Pet))]
    public int PetId { get; set; }

    [JsonIgnore]
    public Pet Pet { get; set; } = null!;

    [ForeignKey(nameof(User))]
    public int UserId { get; set; }

    [JsonIgnore]
    public User User { get; set; } = null!;

    [MaxLength(50)]
    public string Status { get; set; } = "Pending";

    [MaxLength(255)]
    public string? Message { get; set; }

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime? UpdatedAt { get; set; }
}
